app/src/main/java/org/mozilla/fenix/components/PrivateShortcutCreateManager.kt

```diff
@@ -0,0 +1,47 @@
+package org.mozilla.fenix.components
+
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import androidx.core.content.pm.ShortcutInfoCompat
+import androidx.core.content.pm.ShortcutManagerCompat
+import androidx.core.graphics.drawable.IconCompat
+import org.mozilla.fenix.HomeActivity
+import org.mozilla.fenix.R
+import org.mozilla.fenix.home.intent.StartSearchIntentProcessor
+import java.util.UUID
+
+/**
+ * Handles the creation of pinned shortcuts.
+ */
+class PrivateShortcutCreateManager {
+
+    companion object {
+        fun createPrivateShortcut(context: Context) {
+            if (ShortcutManagerCompat.isRequestPinShortcutSupported(context)) {
```

Comment:
nit: consider doing an early return here so all of this code isn't in an indented block:

```
if !(ShortcutManagerCompat.isRequestPinShortcutSupported(context)) { return }
// Rest of the code here
```